docs/source/basics/messages.rst

```diff
@@ -0,0 +1,144 @@
+.. _messages_usage_label:
+
+========
+Messages
+========
+
+Messages are a core component in torchtune that govern how text and multimodal content is tokenized. It serves as the common interface
+for all tokenizer and datasets APIs to operate on. Messages contain information about the text content, which role is sending the text
+content, and other information relevant for special tokens in model tokenizers. For more information about the individual parameters
+for Messages, see the API ref for :class:`~torchtune.data.Message`. Here, we briefly discuss how to create messages, format messages, and
+tokenize messages.
+
+.. _creating_messages:
+
+Creating Messages
+-----------------
+
+Messages can be created via the standard class constructor or directly from a dictionary.
+
+.. code-block:: python
+
+    from torchtune.data import Message
```

Comment:
This is great. Maybe giving a bit of context in the code, showing which class creates or receives a message, i think it would make it easier for the user to understand how things connect end2end.